app/models/category_list.rb

```diff
@@ -135,21 +135,21 @@ def dismissed_topic?(topic)
   end
 
   def find_categories
+    # Enforce paginaion for users who can see a large number of categories to
+    # smooth out the performance of the category list page.
+    paginate =
+      Category.secured(@guardian).count > MAX_UNOPTIMIZED_CATEGORIES ||
+        @guardian.can_lazy_load_categories?
```

Comment:
Can we simplify this even further by paginating as long as the page style is `categories_only`? Even if a site has a low number of categories, there is no harm paginating.

Reply:
It is not the same thing. When pagination is enabled, at most 5 subcategories are loaded for each category.